mdbackup/actions/builtin/command.py

```
@@ -0,0 +1,176 @@
+        args.append('-A')
+    else:
+        args.append('-a')
+
+    if identity_file is not None:
+        args.extend('-i', identity_file)
+
+    if user is not None:
+        args.extend('-l', user)
+
+    if config_file is not None:
+        args.extend('-F', params['configFile'])
+
+    args.extend(['-x', params['host']])
+    if param_args is not None:
+        args.extend(param_args)
+    elif param_command is not None:
+        args.extend(_parse_command(param_command))
+    else:
+        raise KeyError('no args nor command defined')
+
+    return action_command(inp, {'args': args, 'env': env})
+
+
+@action('docker', input='stream', output='stream:process')
+def action_docker(inp, params) -> subprocess.Popen:
+    args = ['docker', 'container', 'run', '--rm', '-i']
+
+    volumes: list = params.get('volumes', [])
+    env = params.get('env', [])
+    user = params.get('user')
+    group = params.get('group')
+    network = params.get('network', 'host')
+    workdir = params.get('workdir')
+    param_args: list = params.get('args')
+    param_command = params.get('command')
+    pull = params.get('pull', False)
+    image = params['image']
+
+    for volume in volumes:
+        args.extend(['-v', volume])
+
+    if isinstance(env, dict):
+        env = map(lambda pair: f'{pair[0]}={pair[1]}', env.items())
+    for env_var in env:
+        args.extend(['-e', env_var])
+
+    if user is not None:
+        args.extend('-u', user)
+        if group is not None:
+            args[-1] = f'{args[-1]}:{group}'
+
+    args.append(f'--network={network}')
+
+    if workdir is not None:
+        args.extend(['-w', workdir])
+
+    if pull is not None:
+        _logger.info(f'Pulling image {image}...')
+        proc = action_command(None, {'command': f'docker image pull -q "{image}"'})
+        stdout, stderr = proc.communicate()
+        if proc.returncode != 0:
+            _logger.error(f'Image pull for {image} failed:\n{stderr.decode("utf-8")}')
```

Should raise an exception to end the action